lib/src/wallet/crypto_amount_format.dart

```diff
@@ -1,28 +1,75 @@
+import 'package:oxen_wallet/generated/l10n.dart';
+
+class AmountDetail {
+  const AmountDetail(this.index, this.fraction);
+
+  final int index;
+  final int fraction;
+
+  static const List<AmountDetail> all = [
+    AmountDetail.ultra,
+    AmountDetail.detailed,
+    AmountDetail.normal,
+    AmountDetail.none
+  ];
+  static const AmountDetail ultra = AmountDetail(0, 9);
+  static const AmountDetail none = AmountDetail(1, 0);
+  static const AmountDetail detailed = AmountDetail(2, 4);
+  static const AmountDetail normal = AmountDetail(3, 2);
```

Comment:
Something doesn't look right here; the first value, I believe, should be the index in `all`, but they are mismatched: `none` is at index `1` but `all[1]` is `AmountDetail.detailed`.  I think these should be:
```dart
  static const AmountDetail ultra = AmountDetail(0, 9);
  static const AmountDetail detailed = AmountDetail(1, 4);
  static const AmountDetail normal = AmountDetail(2, 2);
  static const AmountDetail none = AmountDetail(3, 0);
```

Reply:
Actually, possibly not as the `toString()` below is following the original values here.  (But then I think that means deserialize is doing the wrong thing?)